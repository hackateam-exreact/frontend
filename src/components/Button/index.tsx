import {
  Button as ChakraButton,
  ButtonProps as ChakraButtonProps
} from '@chakra-ui/react'

export interface ButtonProps extends ChakraButtonProps {
  backgroundColor?: 'blue.500' | 'green.500'
  children: string
  fullWidth?: boolean
  size?: 'xs' | 'sm' | 'md' | 'lg'
}

export const Button = ({
  backgroundColor = 'blue.500',
  children,
  fullWidth = false,
  size = 'md',
  ...props
}: ButtonProps) => (
  <ChakraButton
    color="white"
    backgroundColor={backgroundColor}
    width={fullWidth ? '100%' : 'auto'}
    size={size}
    _hover={{
      backgroundColor
    }}
    {...props}
  >
    {children}
  </ChakraButton>
)
