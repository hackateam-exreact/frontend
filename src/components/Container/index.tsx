import { Flex, FlexProps, useColorMode } from '@chakra-ui/react'

export const Container = (props: FlexProps) => {
  const { colorMode } = useColorMode()

  const bgColor = { light: 'gray.50', dark: 'black.500' }
  const color = { light: 'black', dark: 'white.500' }

  return (
    <Flex
      direction="column"
      alignItems="center"
      justifyContent="flex-start"
      bg={bgColor[colorMode]}
      color={color[colorMode]}
      {...props}
    />
  )
}
