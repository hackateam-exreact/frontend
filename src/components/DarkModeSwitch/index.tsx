import { useColorMode, Switch } from '@chakra-ui/react'

const DarkModeSwitch = () => {
  const { colorMode, toggleColorMode } = useColorMode()
  const isDark = colorMode === 'dark'

  return (
    <Switch
      position="fixed"
      top="1rem"
      right="1rem"
      aria-label="toggle color mode"
      isChecked={isDark}
      onChange={toggleColorMode}
    />
  )
}

export default DarkModeSwitch
