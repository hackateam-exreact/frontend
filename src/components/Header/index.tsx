import { Link as ChakraLink, Flex, HStack, Image } from '@chakra-ui/react'

import Link from 'next/link'
import React from 'react'
import { Searchbar } from './Searchbar'
import { UserBadge } from 'components/Profile/UserBadge'
import { useAuth } from 'hooks/useAuth'
import { useRouter } from 'next/router'

export function Header() {
  const { user } = useAuth()
  const { asPath } = useRouter()

  const blacklist = ['/', '/dev/signin', '/dev/signup', '/dev/welcome']

  return (
    <>
      {!blacklist.includes(asPath) && (
        <Flex
          w="100%"
          h="6rem"
          borderBottom="1px"
          borderColor="gray.800"
          mb="1rem"
        >
          <HStack
            w="100%"
            maxW="1400px"
            mx="auto"
            align="center"
            spacing="10"
            px="5"
          >
            <Link href="/" passHref={true}>
              <ChakraLink>
                <Image
                  src="/img/logo.png"
                  alt="Devspot"
                  w="225px"
                  objectFit="cover"
                />
              </ChakraLink>
            </Link>
            <Searchbar />
            <UserBadge user={user} badgeSide="right" />
          </HStack>
        </Flex>
      )}
    </>
  )
}
