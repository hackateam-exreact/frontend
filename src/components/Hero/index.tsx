import { Flex, Heading } from '@chakra-ui/react'

export type HeroProps = {
  title?: string
}

export const Hero = ({ title = 'Fullstack Alchemists' }: HeroProps) => (
  <Flex
    justifyContent="center"
    alignItems="center"
    height="100vh"
    bgGradient="linear(to-l, #7928CA, #FF0080)"
    bgClip="text"
  >
    <Heading fontSize="6vw">{title}</Heading>
  </Flex>
)
