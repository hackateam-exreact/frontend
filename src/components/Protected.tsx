import { ReactNode, useEffect, useState } from 'react'

import { useAuth } from 'hooks/useAuth'

interface ProtectedProps {
  needsAuthentication?: boolean
  needsAuthorization?: boolean
  children: ReactNode
}

export function Protected({
  needsAuthentication = true,
  needsAuthorization = false,
  children
}: ProtectedProps) {
  const { isAuthenticated, isAuthorized } = useAuth()
  const [canShow, setCanShow] = useState(false)

  const handleCheckPermissions = () => {
    if (needsAuthentication && needsAuthorization) {
      if (isAuthenticated && isAuthorized) return true
    }

    if (needsAuthentication) {
      if (isAuthenticated) return true
    }

    if (needsAuthorization) {
      if (isAuthorized) return true
    }

    return false
  }

  useEffect(() => {
    setCanShow(handleCheckPermissions())
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [isAuthenticated, isAuthorized])

  return <>{canShow && children}</>
}
