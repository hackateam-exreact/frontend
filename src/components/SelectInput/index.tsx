import {
  Select as ChakraSelect,
  FormControl,
  FormErrorMessage,
  FormLabel,
  SelectProps
} from '@chakra-ui/react'
import { ForwardRefRenderFunction, forwardRef } from 'react'

import { FieldError } from 'react-hook-form'
import { ReactNode } from 'react'

export interface SelectInputPropt extends SelectProps {
  children: ReactNode
  inputName: string
  label?: string
  error?: FieldError
  focusBorderColor?: 'blue.500' | 'green.500'
}

const SelectInputBase: ForwardRefRenderFunction<
  HTMLSelectElement,
  SelectInputPropt
> = (
  {
    children,
    inputName,
    label,
    error = null,
    focusBorderColor = 'blue.500',
    ...rest
  },
  ref
) => {
  return (
    <FormControl isInvalid={!!error}>
      {label && <FormLabel htmlFor={inputName}>{label}</FormLabel>}
      <ChakraSelect
        name={inputName}
        id={inputName}
        focusBorderColor={focusBorderColor}
        errorBorderColor="red.500"
        bg="gray.800"
        border="none"
        size="lg"
        _placeholder={{ fontSize: 'sm' }}
        ref={ref}
        {...rest}
      >
        {children}
      </ChakraSelect>
      {!!error && (
        <FormErrorMessage color="red.500">{error.message}</FormErrorMessage>
      )}
    </FormControl>
  )
}

export const SelectInput = forwardRef(SelectInputBase)
