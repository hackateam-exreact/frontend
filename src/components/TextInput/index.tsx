import {
  Input as ChakraInput,
  InputProps as ChakraInputProps,
  FormControl,
  FormErrorMessage,
  FormLabel,
  InputGroup,
  InputLeftElement,
  InputRightElement
} from '@chakra-ui/react'
import { ForwardRefRenderFunction, forwardRef } from 'react'

import { FieldError } from 'react-hook-form'
import { ReactElement } from 'react'

export interface InputProps extends ChakraInputProps {
  inputName: string
  label?: string
  error?: FieldError
  focusBorderColor?: 'blue.500' | 'green.500'
  rightIcon?: ReactElement
  leftIcon?: ReactElement
}

const TextInputBase: ForwardRefRenderFunction<HTMLInputElement, InputProps> = (
  {
    inputName,
    label,
    error = null,
    focusBorderColor = 'blue.500',
    rightIcon = null,
    leftIcon = null,
    ...rest
  },
  ref
) => {
  return (
    <FormControl isInvalid={!!error}>
      {label && <FormLabel htmlFor={inputName}>{label}</FormLabel>}
      <InputGroup>
        {!!leftIcon && (
          <InputLeftElement pointerEvents="all">{leftIcon}</InputLeftElement>
        )}
        <ChakraInput
          name={inputName}
          id={inputName}
          focusBorderColor={focusBorderColor}
          errorBorderColor="red.500"
          bg="gray.800"
          border="none"
          p="3"
          size="lg"
          _placeholder={{ fontSize: 'sm' }}
          ref={ref}
          {...rest}
        />
        {!!rightIcon && (
          <InputRightElement pointerEvents="all">{rightIcon}</InputRightElement>
        )}
      </InputGroup>
      {!!error && (
        <FormErrorMessage color="red.500">{error.message}</FormErrorMessage>
      )}
    </FormControl>
  )
}

export const TextInput = forwardRef(TextInputBase)
