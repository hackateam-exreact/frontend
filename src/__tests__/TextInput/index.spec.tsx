import { render, screen } from 'utils/test-utils'

import { FiMail } from 'react-icons/fi'
import { TextInput } from '../../components/TextInput'

describe('<TextInput />', () => {
  it('should render with label', () => {
    const { container } = render(<TextInput label="Label" inputName="email" />)

    expect(screen.getByLabelText('Label')).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render with a left icon', () => {
    render(
      <TextInput
        leftIcon={<FiMail data-testid="icon" />}
        label="Label"
        inputName="email"
      />
    )

    expect(screen.getByTestId('icon')).toBeInTheDocument()
  })

  it('should render with a right icon', () => {
    render(
      <TextInput
        rightIcon={<FiMail data-testid="icon" />}
        label="Label"
        inputName="email"
      />
    )

    expect(screen.getByTestId('icon')).toBeInTheDocument()
  })

  it('should render with an error message', () => {
    render(
      <TextInput
        leftIcon={<FiMail data-testid="icon" />}
        label="Label"
        inputName="email"
        error={{
          message: 'Invalid E-mail',
          type: 'maxLength'
        }}
      />
    )

    expect(screen.getByText('Invalid E-mail')).toBeInTheDocument()
  })

  it('should not render input group if there is no icon props', () => {
    render(<TextInput label="Label" inputName="email" />)

    expect(screen.queryByTestId('input-group')).toBeNull()
  })
})
