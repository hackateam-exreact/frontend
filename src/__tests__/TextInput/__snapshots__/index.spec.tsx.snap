// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> should render with label 1`] = `
.emotion-0 {
  width: 100%;
  position: relative;
}

.emotion-1 {
  display: block;
  text-align: start;
  font-size: var(--chakra-fontSizes-md);
  -webkit-margin-end: var(--chakra-space-3);
  margin-inline-end: var(--chakra-space-3);
  margin-bottom: var(--chakra-space-2);
  font-weight: var(--chakra-fontWeights-medium);
  -webkit-transition: all 0.2s;
  transition: all 0.2s;
  opacity: 1;
}

.emotion-1[disabled],
.emotion-1[aria-disabled=true],
.emotion-1[data-disabled] {
  opacity: 0.4;
}

.emotion-2 {
  width: 100%;
  min-width: 0px;
  outline: 2px solid transparent;
  outline-offset: 2px;
  position: relative;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  -webkit-transition: all 0.2s;
  transition: all 0.2s;
  font-size: var(--chakra-fontSizes-lg);
  -webkit-padding-start: var(--chakra-space-4);
  padding-inline-start: var(--chakra-space-4);
  -webkit-padding-end: var(--chakra-space-4);
  padding-inline-end: var(--chakra-space-4);
  height: var(--chakra-sizes-12);
  border-radius: var(--chakra-radii-md);
  border: var(--chakra-borders-none);
  border-color: inherit;
  background: var(--chakra-colors-gray-800);
  padding: var(--chakra-space-3);
}

.emotion-2:hover,
.emotion-2[data-hover] {
  border-color: var(--chakra-colors-whiteAlpha-400);
}

.emotion-2[aria-readonly=true],
.emotion-2[readonly],
.emotion-2[data-readonly] {
  box-shadow: none!important;
  -webkit-user-select: all;
  -moz-user-select: all;
  -ms-user-select: all;
  user-select: all;
}

.emotion-2[disabled],
.emotion-2[aria-disabled=true],
.emotion-2[data-disabled] {
  opacity: 0.4;
  cursor: not-allowed;
}

.emotion-2[aria-invalid=true],
.emotion-2[data-invalid] {
  border-color: #FF4062;
  box-shadow: 0 0 0 1px #FF4062;
}

.emotion-2:focus,
.emotion-2[data-focus] {
  z-index: 1;
  border-color: #5271FF;
  box-shadow: 0 0 0 1px #5271FF;
}

.emotion-2::-webkit-input-placeholder {
  font-size: var(--chakra-fontSizes-sm);
}

.emotion-2::-moz-placeholder {
  font-size: var(--chakra-fontSizes-sm);
}

.emotion-2:-ms-input-placeholder {
  font-size: var(--chakra-fontSizes-sm);
}

.emotion-2::placeholder {
  font-size: var(--chakra-fontSizes-sm);
}

<div
  class="chakra-form-control emotion-0"
  role="group"
>
  <label
    class="chakra-form__label emotion-1"
    for="email"
    id="field-1-label"
  >
    Label
  </label>
  <input
    class="chakra-input emotion-2"
    id="email"
    name="email"
  />
</div>
`;
